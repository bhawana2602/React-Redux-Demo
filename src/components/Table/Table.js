import React from "react";
import styled from "styled-components";

const TH = styled.th`
  padding: 10px;
  background: #269faf66;
`;
const TD = styled.td`
  padding: 8px;
`;

const Table = ({ userObj }) => {
  console.log(userObj, "functional component");
  return (
    <table border="1" className="table">
      <thead>
        <TH>First Name</TH>
        <TH>Last Name</TH>
        <TH>Email</TH>
        <TH>Profile</TH>
      </thead>
      <tbody>
        {userObj.data &&
          userObj.data.data.map((x, i) => (
            <tr key={i}>
              <TD>{x.first_name}</TD>
              <TD>{x.last_name}</TD>
              <TD>{x.email}</TD>
              <TD>
                <img src={x.avatar} width="40" height="40" alt="profile" />
              </TD>
            </tr>
          ))}
      </tbody>
    </table>
  );
};

export default Table;
