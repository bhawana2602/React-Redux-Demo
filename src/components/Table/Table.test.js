import React from "react";
import Table from "./Table";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Table Component", () => {
  let props;

  beforeEach(() => {
    props = {
      data: {
        page: 1,
        per_page: 6,
        total: 1,
        total_pages: 1,
        data: [
          {
            id: 1,
            email: "byron.fields@reqres.in",
            first_name: "Byron",
            last_name: "Fields",
            avatar:
              "https://s3.amazonaws.com/uifaces/faces/twitter/russoedu/128.jpg",
          },
        ],
        ad: {
          company: "StatusCode Weekly",
          url: "http://statuscode.org/",
          text: "demo text.",
        },
      },
      loading: false,
      error: null,
    };
  });

  it("renders with data", () => {
    const tree = renderer.create(<Table userObj={props} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
