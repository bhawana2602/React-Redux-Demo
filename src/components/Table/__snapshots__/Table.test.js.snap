// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders with data 1`] = `
<table
  border="1"
  className="table"
>
  <thead>
    <th
      className="sc-AxjAm bKEskS"
    >
      First Name
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Last Name
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Email
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Profile
    </th>
  </thead>
  <tbody>
    <tr>
      <td
        className="sc-AxirZ dKMvOG"
      >
        Byron
      </td>
      <td
        className="sc-AxirZ dKMvOG"
      >
        Fields
      </td>
      <td
        className="sc-AxirZ dKMvOG"
      >
        byron.fields@reqres.in
      </td>
      <td
        className="sc-AxirZ dKMvOG"
      >
        <img
          alt="profile"
          height="40"
          src="https://s3.amazonaws.com/uifaces/faces/twitter/russoedu/128.jpg"
          width="40"
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Component renders with no data 1`] = `
<table
  border="1"
  className="table"
>
  <thead>
    <th
      className="sc-AxjAm bKEskS"
    >
      First Name
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Last Name
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Email
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Profile
    </th>
  </thead>
  <tbody>
    <tr>
      <td
        className="sc-AxirZ dKMvOG"
      >
        Byron
      </td>
      <td
        className="sc-AxirZ dKMvOG"
      >
        Fields
      </td>
      <td
        className="sc-AxirZ dKMvOG"
      >
        byron.fields@reqres.in
      </td>
      <td
        className="sc-AxirZ dKMvOG"
      >
        <img
          alt="profile"
          height="40"
          src="https://s3.amazonaws.com/uifaces/faces/twitter/russoedu/128.jpg"
          width="40"
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Component renders without crashing 1`] = `
<table
  border="1"
  className="table"
>
  <thead>
    <th
      className="sc-AxjAm bKEskS"
    >
      First Name
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Last Name
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Email
    </th>
    <th
      className="sc-AxjAm bKEskS"
    >
      Profile
    </th>
  </thead>
  <tbody />
</table>
`;
